.github/workflows/ci.yaml

```diff
@@ -1,28 +1,51 @@
 name: CI
-on: push
+on: [push, pull_request]
 
 jobs:
   rspec:
+    permissions:
+      contents: read
+      checks: write
+      pull-requests: write
+      statuses: write
     runs-on: ubuntu-latest
     container: public.ecr.aws/docker/library/ruby:3.2.2
     steps:
       - name: Checkout
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 100
       - name: Install Deps
         run: |
-          apt update && apt install ffmpeg -y
+          apt update && apt install ffmpeg cmake -y
           bundle
       - name: RSpec
         run: |
           bundle exec rspec
+      - name: Setup Pronto
+        if: github.event_name == 'pull_request'
+        run: gem install pronto pronto-rubocop
+      - name: Run Pronto
+        if: github.event_name == 'pull_request'
+        run: |
+          set -x
+          bundle install
+          git config --global --add safe.directory $GITHUB_WORKSPACE
+          git fetch --no-tags --prune --depth=100 origin ${{ github.base_ref }}
+          bundle exec pronto run -f github_status github_pr -c origin/${{ github.base_ref }}
+        env:
+          PRONTO_PULL_REQUEST_ID: ${{ github.event.pull_request.number }}
+          PRONTO_GITHUB_ACCESS_TOKEN: "${{ github.token }}"
+          BUNDLE_PATH: vendor/bundle
   lint:
     runs-on: ubuntu-latest
     container: public.ecr.aws/docker/library/ruby:3.2.2
     steps:
       - name: Checkout
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
       - name: Install Deps
         run: |
+          apt update && apt install cmake -y
```

Comment:
Do we need to install `cmake` for rubocop to run? 🤔

Reply:
`cmake` is for rugged - which is used by Pronto/Undercover to perform git operations